deptry/utils.py

```diff
@@ -39,3 +41,64 @@ def load_pyproject_toml(pyproject_toml_path: str = PYPROJECT_TOML_PATH) -> Dict[
             return tomllib.load(pyproject_file)
     except FileNotFoundError:
         raise FileNotFoundError(f"No file `pyproject.toml` found in directory {os.getcwd()}")
+
+
+def find_site_packages_below(root: Path, platform: str) -> Optional[Path]:
+    """Find site packages directory below a virtual environment root"""
+    if platform == "Windows":
+        pattern = "Lib/site-packages"
+    else:
+        pattern = "lib/python*/site-packages"
+    try:
+        return next(root.rglob(pattern))
+    except StopIteration:
+        return None
+
+
+def in_project_virtualenv(
+    project_name: str, *, prefix: str, base_prefix: str, active_virtual_env: Optional[str] = None
```

Comment:
See above. If possible it would be good if we can reduce the [number of arguments](https://matheus.ro/2018/01/29/clean-code-avoid-many-arguments-functions/). Do you have any ideas on how to achieve that?

Reply:
So my main motivation is to make the function easier to unit test. If I called `sys.prefix` and co from within I would need quite a lot of `mock`ing and I eschew that 😅 But a compromise can be found: I could implement an `ExecutionContext` dataclass that wraps all the arguments, with a `ExecutionContext.from_runtime(project_name)` constructor. And pass this to the function. What do you think ?

Reply:
That sounds like a good proposal, that makes this function-call a lot more clear!

Reply:
Implemented in [3867c1d](https://github.com/fpgmaas/deptry/pull/207/commits/3867c1dd45db69c72ca768b5fbceabe27733afe2), and when running_in_virtualenv was left with a single ctx argument, I thought I might as well call it self sweat_smile